src/cephadm/cephadm

```diff
@@ -7099,28 +7112,41 @@ def authorize_ssh_key(ssh_pub_key: str, ssh_user: str) -> None:
             f.write('\n')
         f.write(ssh_pub_key + '\n')
 
+    return True
+
 
 def revoke_ssh_key(key: str, ssh_user: str) -> None:
     """Revoke the public key authorization for the ssh user"""
-    _, _, ssh_dir = get_ssh_vars(ssh_user)
+    ssh_uid, ssh_gid, ssh_dir = get_ssh_vars(ssh_user)
     auth_keys_file = '%s/authorized_keys' % ssh_dir
     deleted = False
     if os.path.exists(auth_keys_file):
         with open(auth_keys_file, 'r') as f:
             lines = f.readlines()
-        with open(auth_keys_file, 'w') as f:
+        _, filename = tempfile.mkstemp()
+        with open(filename, 'w') as f:
+            os.fchown(f.fileno(), ssh_uid, ssh_gid)
+            os.fchmod(f.fileno(), 0o600)  # secure access to the keys file
```

Comment:
while 600 permissions do make sense for this file, I don't think we'd want to overwrite the permissions the user is currently using for their authorized_keys file. Can we detect what file permissions they're using and re-use that?